src/v/storage/log_manager.cc

```diff
@@ -198,6 +200,30 @@ log_manager::create_cache(with_cache ntp_cache_enabled) {
     return batch_cache_index(_batch_cache);
 }
 
+/*
+ * Clears the `_logs` container while giving seastar the ability to yield the
+ * fiber after every log is erased. This is used to avoid reactor stall warnings
+ * when deleting `_logs` when it contains a large number of objects.
+ *
+ * Note this function should only be called once `_logs` is no longer
+ * being used as it doesn't check for iterator invalidation.
+ */
+ss::future<> log_manager::async_clear_logs() {
+    constexpr size_t threshold_size = 1000;
+
+    // Avoid excessive erases on small containers.
+    if (_logs.size() <= threshold_size) {
+        _logs.clear();
+        co_return;
+    }
+
+    auto current = _logs.begin();
+    while (current != _logs.end()) {
+        current = _logs.erase(current, ++current);
+        co_await ss::coroutine::maybe_yield();
```

Comment:
@ballard26 what other places in the core have this pattern, maybe the batch cache?

Reply:
also @ballard26 maybe shadow-indexing has this issue too - effectively anything that deals w/ cleanup?

Reply:
Yeah, potentially any unbounded container could stall a reactor during it's destruction. In this case `_logs` grows linearly with the number of partitions and eventually we reached to point where it's destruction took longer than seastar's stall timeout.

Reply:
maybe worth skimming some other knonw code paths and filing issues to keep track of them. not a blocker for this pr.